### PR TITLE
Code Generation - Simplify completion pattern

### PR DESCRIPTION
**Pull request by diesalbla**
We can remove the use of an `isInstanceOf` clause and instead use a typed pattern

### CODE REVIEW

File: continuationsPlugin/src/main/scala/continuations/DefDefTransforms.scala [hunk 1]
```diff
@@ -863,32 +863,25 @@ object DefDefTransforms extends TreesChecks:
...
-              .select(labelVarParam)
-              .select(integerAND)
-              .appliedTo(integerMin)
-              .select(integerNE)
-              .appliedTo(tpd.Literal(Constant(0x0)))),
+          .select(labelVarParam)
+          .select(integerAND)
+          .appliedTo(integerMin)
+          .select(integerNE)
+          .appliedTo(tpd.Literal(Constant(0x0))),
```
Comment (jackcviers) [inline continuationsPlugin/src/main/scala/continuations/DefDefTransforms.scala:879]: 👍🏻

File: continuationsPlugin/src/test/scala/continuations/StateMachineFixtures.scala [hunk 2]
```diff
@@ -1149,22 +1149,16 @@ trait StateMachineFixtures {
...
        |                      compileFromString$package$foo$1
        |                  ].$label.&(scala.Int.MinValue).!=(0)
-       |                )
        |               =>
-       |                $continuation =
-       |                  x$0.asInstanceOf[
-       |                    continuations.compileFromString$package.
-       |                      compileFromString$package$foo$1
-       |                  ]
+       |                $continuation =|                  x$0
```
Comment (jackcviers) [inline continuationsPlugin/src/test/scala/continuations/StateMachineFixtures.scala:1161]: This pipe looks like a typo:

```suggestion
       |                $continuation = x$0
```

**Review by jackcviers — CHANGES_REQUESTED** [review]

**Review by jackcviers — CHANGES_REQUESTED** [review]
Also needs a scalafmtAll

**Comment by gregorath** [comment]
yeah I remember I spent quite some time to make this work like this but annoyingly didn't manage to. it looks like it fails in the `GenBCode` phase. happy to help you debug it if you need to see the tree that fails etc. it would be nice if we manage to do it.

**Review by jackcviers — APPROVED** [review]

**Comment by diesalbla** [comment]
An update on this work: yesterday @gregorath and I started looking at the trees generated by the compiler phases. We found the problem to be in the pattern-matcher. Before this phase, the body . However, within the pattern matcher the variable `x$0` is renamed to a fresh variable `x1`, but _only in the pattern_ itself, not in the body of the first `case`.

File: continuationsPlugin/src/test/scala/continuations/ContinuationsPluginSuite.scala [hunk 3]
```diff
@@ -135,19 +135,27 @@ class ContinuationsPluginSuite extends FunSuite, CompilerFixtures, StateMachineF
       }
   }
 
-  compilerContext.test("debug".ignore) {
+  compilerContextWithContinuationsPlugin.test("debug".only) {
```
Comment (jackcviers) [inline continuationsPlugin/src/test/scala/continuations/ContinuationsPluginSuite.scala:138]: You have to remove the .only to get reports on everything in the checks.

Comment (jackcviers) [reply continuationsPlugin/src/test/scala/continuations/ContinuationsPluginSuite.scala:138]: We should be able to remove the debug test entirely, we have a better handle on how to get the tree outputs at need now.

**Comment by diesalbla** [comment]
> An update on this work: yesterday @gregorath and I started looking at the trees generated by the compiler phases. We found the problem to be in the pattern-matcher. Before this phase, the body . However, within the pattern matcher the variable x$0 is renamed to a fresh variable x1, but only in the pattern itself, not in the body of the first case.

It seems that by adding an extra wrapper of a `Bind`, with the `Typed` on the right hand side, it now works fine and recognises as a variable (waiting for CI)

**Pull request merged**